eq-author/src/App/page/Design/Validation/AnswerValidation.js

```diff
@@ -179,180 +166,88 @@ class AnswerValidation extends React.PureComponent {
 
-    let validationButtons = [];
+  if (validValidationTypes.length === 0) {
+    return null;
+  }
 
-    if (answer.type === "DateRange") {
-      const { duration, range } = validValidationTypes.reduce(
-        (accumulator, current) => {
-          const { id } = current;
-          if (id === "minDuration" || id === "maxDuration") {
-            if (!accumulator.duration) {
-              accumulator.duration = [];
-            }
-            accumulator.duration.push(current);
-          }
-          if (id === "earliestDate" || id === "latestDate") {
-            if (!accumulator.range) {
-              accumulator.range = [];
-            }
-            accumulator.range.push(current);
-          }
-          return accumulator;
-        },
-        {}
-      );
+  const validationErrors = [];
+  let validationButtons = [];
 
-      const durationButtons = this.renderValidation(
-        duration,
-        answer,
-        validationErrors
-      );
+  validValidationTypes.forEach(type => {
+    let validationMessage;
 
-      const rangeButtons = this.renderValidation(
-        range,
-        answer,
-        validationErrors
-      );
+    const validation = get(answer, `validation.${type.id}`, {});
+    const errors = get(validation, `validationErrorInfo.errors`, []);
+    const { enabled, inclusive } = validation;
+    const value = enabled ? type.preview(validation, answer) : null;
 
-      const durationError = this.renderPropertyError(
-        validationErrors.dateRange.duration.length > 1,
-        DURATION_ERROR_MESSAGE,
-        "duration-error"
-      );
+    const onClick = () => {
+      setModalIsOpen(true);
+      setStartingTabId(type.id);
+    };
 
-      const rangeError = this.renderPropertyError(
-        validationErrors.dateRange.range.length > 1,
-        EARLIEST_BEFORE_LATEST_DATE,
-        "range-error"
-      );
+    validationButtons.push(
+      <ValidationButton id={type.id} hasError={errors.length > 0} onClick={onClick} title={type.title}>
+        <Title>{titleText(type.id, type.title, enabled, inclusive)}</Title>
+        {enabled && !isNull(value) && <Detail>{value}</Detail>}
+      </ValidationButton>
+    );
 
-      validationButtons.push(rangeButtons);
-      validationButtons.push(rangeError);
+    if (type.id === "earliestDate" || type.id === "minDuration") {
+      return; // Don't display anything after the earliest date / min duration buttons - show after section
+    }
 
-      validationButtons.push(durationButtons);
-      validationButtons.push(durationError);
+    if (type.id === "maxDuration") {
+      validationMessage = DURATION_ERROR_MESSAGE;
+    }
+    else if (type.id === "latestDate") {
+      validationMessage = EARLIEST_BEFORE_LATEST_DATE;
+    }
+    else if (answer.type === "Date") {
+      validationMessage = EARLIEST_BEFORE_LATEST_DATE;
+    }
+    else if (find(validationErrors, error => error.errorCode.include("ERR_NO_VALUE"))) {
```

Comment:
```suggestion
    else if (find(validationErrors, ({errorCode}) => errorCode.include("ERR_NO_VALUE"))) {
```

I only found this out a few weeks ago, but you can destructure objects as they're being used as well as when assigning them to variables.